tardis/transport/montecarlo/packet_trackers.py

```diff
@@ -103,6 +144,68 @@ def track(self, r_packet):
         ] = r_packet.last_interaction_type
         self.num_interactions += 1
 
+    def track_boundary_interaction(self, current_shell_id, next_shell_id):
+        if (
+            self.num_boundary_interactions
+            >= self.boundary_interaction_array_length
+        ):
+            temp_length = self.boundary_interaction_array_length * 2
+
+            temp_boundary_interaction = np.empty(
+                temp_length, dtype=boundary_interaction_dtype
+            )
+            temp_boundary_interaction[
+                : self.boundary_interaction_array_length
+            ] = self.boundary_interaction
+
+            self.boundary_interaction = temp_boundary_interaction
+            self.boundary_interaction_array_length = temp_length
+
+        self.boundary_interaction[self.num_boundary_interactions][
+            "event_id"
+        ] = self.event_id
+        self.boundary_interaction[self.num_boundary_interactions][
+            "current_shell_id"
+        ] = current_shell_id
+        self.boundary_interaction[self.num_boundary_interactions][
+            "next_shell_id"
+        ] = next_shell_id
+        self.num_boundary_interactions += 1
+        self.event_id += 1
+
+    def track_line_interaction(self, r_packet):
+        if self.num_line_interactions >= self.line_interaction_array_length:
```

Comment:
I think this `if` block, and the one in `track_boundary_interaction`, may be better as a separate method that performs the array extension.

Reply:
Yes. That makes sense. I will look into it.

Reply:
Can they both be merged into a single `extend` function?

Reply:
That would be ideal.